src/toil/wdl/wdltoil.py

```diff
@@ -470,11 +485,21 @@ def _devirtualize_filename(self, filename: str) -> str:
         if filename.startswith(TOIL_URI_SCHEME):
             # This is a reference to the Toil filestore.
             # Deserialize the FileID
-            file_id, file_basename = unpack_toil_uri(filename)
+            file_id, parent_id, file_basename = unpack_toil_uri(filename)
 
             # Decide where it should be put
-            file_dir = self._file_store.getLocalTempDir()
-            dest_path = os.path.join(file_dir, file_basename)
+            if self._is_uuid(parent_id):
+                # This is a URI with the "parent" UUID attached to the filename
+                # Use UUID as folder name rather than a new temp folder to reduce internal clutter
+                if not os.path.exists(parent_id):
+                    os.mkdir(parent_id)
+                # Put the UUID in the destination path in order for tasks to see where to put files depending on their parents
+                dest_path = os.path.join(self._file_store.localTempDir, parent_id, file_basename)
+            else:
+                # In case a file was not virtualized with a UUID
+                # But this shouldn't happen
```

Comment:
Why not make the UUID mandatory in `pack_toil_uri`? If it should always be there, we probably want to actually enforce that. We could also move all the type conversion into the pack/unpack functions, so only real UUID objects go in and out.